source/track_thick.f90

```diff
@@ -1886,317 +1886,324 @@ subroutine synuthck
-          aek(j)=abs(ekv(j,l)/dpd(j))
-          hi(j)=sqrt(aek(j))
-          fi(j)=hi(j)*el(l)
-          si(j)=sin_mb(fi(j))
-          co(j)=cos_mb(fi(j))
-          al(1,ih2,j,l)=co(j)
-          al(2,ih2,j,l)=si(j)/hi(j)
-          al(3,ih2,j,l)=(-one*si(j))*hi(j)                           !hr01
-          al(4,ih2,j,l)=co(j)
-          as(4,ih2,j,l)=(((-one*rvv(j))*al(2,ih2,j,l))*al(3,ih2,j,l))/c2e3 !hr01
-          as(5,ih2,j,l)=(((-one*rvv(j))*(el(l)-al(1,ih2,j,l)*al(2,ih2,j,l)))*aek(j))/c4e3
-          as(6,ih2,j,l)=((-one*rvv(j))*(el(l)+al(1,ih2,j,l)*al(2,ih2,j,l)))/c4e3
-        endif
-130     continue
+        if(fok > pieni) then
+          hp = exp_mb(fi)
+          hm = one/hp
+          hc = (hp+hm)*half
+          hs = (hp-hm)*half
+          al(1,ih1,j,l) = hc
+          al(2,ih1,j,l) = hs/hi
+          al(3,ih1,j,l) = hs*hi
+          al(4,ih1,j,l) = hc
+
+          wfa  = ((wf/afok)*(one-hc))/dpsq(j)
+          wfhi = ((wf/hi)*hs)/dpsq(j)
+          al(5,ih1,j,l) = (wfa*dpsv(j))*c1e3
+          al(6,ih1,j,l) = ((-one*wfhi)*dpsv(j))*c1e3
+
+          sm12 = el(l)-al(1,ih1,j,l)*al(2,ih1,j,l)
+          sm23 = al(2,ih1,j,l)*al(3,ih1,j,l)
+          as(1,ih1,j,l) = (((rvv(j)*((dpsv(j)**2/(four*dpd(j)))*sm12&
+            +dpsv(j)*(el(l)-al(2,ih1,j,l))))/afok)*wf**2+el(l)*(one-rvv(j)))*c1e3
+          as(2,ih1,j,l) = (-one*rvv(j))*(((dpsv(j)*wf)/(two*dpsq(j)))*sm12-dpd(j)*wfhi)
+          as(3,ih1,j,l) = rvv(j)*(((((dpsv(j)*half)/afok)/dpd(j))* ed(l))*sm23-dpd(j)*wfa)
+          as(4,ih1,j,l) = ((-one*rvv(j))*sm23)/c2e3
+          as(5,ih1,j,l) = ((rvv(j)*sm12)*afok)/c4e3
+          as(6,ih1,j,l) = ((-one*rvv(j))*(el(l)+al(1,ih1,j,l)*al(2,ih1,j,l)))/c4e3
+
+          aek = abs(ekv(j,l)/dpd(j))
+          hi  = sqrt(aek)
+          fi  = hi*el(l)
+          si  = sin_mb(fi)
+          co  = cos_mb(fi)
+          al(1,ih2,j,l) = co
+          al(2,ih2,j,l) = si/hi
+          al(3,ih2,j,l) = (-one*si)*hi
+          al(4,ih2,j,l) = co
+          as(4,ih2,j,l) = (((-one*rvv(j))*al(2,ih2,j,l))*al(3,ih2,j,l))/c2e3 !hr01
+          as(5,ih2,j,l) = (((-one*rvv(j))*(el(l)-al(1,ih2,j,l)*al(2,ih2,j,l)))*aek)/c4e3
+          as(6,ih2,j,l) = ((-one*rvv(j))*(el(l)+al(1,ih2,j,l)*al(2,ih2,j,l)))/c4e3
+        end if
+      end do
       goto 160
-    elseif (kz1.eq.9) then
+    elseif(kz1 == 9) then
 !-----------------------------------------------------------------------
 !  EDGE FOCUSSING
 !-----------------------------------------------------------------------
-140     do 150 j=1,napx
-        rhoi(j)=ed(l)/dpsq(j)
-        fok(j)=rhoi(j)*tan_mb((el(l)*rhoi(j))*half)                  !hr01
-        al(3,1,j,l)=fok(j)
-        al(3,2,j,l)=-fok(j)
-150     continue
+140   do j=1,napx
+        rhoi = ed(l)/dpsq(j)
+        fok  = rhoi*tan_mb((el(l)*rhoi)*half)                  !hr01
+        al(3,1,j,l) = fok
+        al(3,2,j,l) = -fok
+      end do
       goto 160
     else
 !Eric
 ! Is really an error but old code went to 160
```

Comment:
If it is an error, then print a warning at least? Or the comment is wrong and kz can be not in the range of the else-ifs...

Reply:
Feel free to investigate whether Eric is correct, and add this. The thick tracking code is not well maintained, and not all that well covered in tests. Although most of this routine is covered.